rules/experimental/rbe/exec_properties.bzl

```diff
@@ -156,3 +246,107 @@ def merge_dicts(*dict_args):
     for dictionary in dict_args:
         result.update(dictionary)
     return result
+
+def _exec_property_sets_repository_impl(repository_ctx):
+    repository_ctx.file(
+        "BUILD",
+        content = "",
+        executable = False,
+    )
+    repository_ctx.file(
+        "constants.bzl",
+        content = repository_ctx.attr.constants_bzl_content,
+        executable = False,
+    )
+
+# _exec_property_sets_repository is a repository rule that creates a repo with the specified exec_property_sets.
+_exec_property_sets_repository = repository_rule(
+    implementation = _exec_property_sets_repository_impl,
+    local = True,
+    attrs = {
+        "constants_bzl_content": attr.string(
+            mandatory = True,
+            doc = "The content of the constants.bzl file within the repository rule.",
+        ),
+    },
+)
+
+def _verify_dict_of_dicts(name, dicts):
+    """ Verify that dict is of type {string->{string->string}}.
+
+    Args:
+      name: Name of the repo rule. Used for error messages.
+      dicts: a dict whose key is a string and whose value is a dict from string to string.
+    """
+
+    # Verify that dict is of type {string->{string->string}}.
+    for key, value in dicts.items():
+        if type(key) != "string":
+            fail("In repo rule %s, execution property set name %s must be a string" % (name, key))
+        if type(value) != "dict":
+            fail("In repo rule %s, execution property set of %s must be a dict" % (name, key))
+        for k, v in value.items():
+            if type(k) != "string":
+                fail("In repo rule %s, execution property set %s, the key %s must be a string" % (name, key, k))
+            if type(v) != "string":
+                fail("In repo rule %s, execution property set %s, key %s, the value %s must be a string" % (name, key, k, v))
+
+def custom_exec_properties(name, dicts):
+    """ Creates a repository containing execution property dicts.
+
+    Use this macro in your WORKSPACE.
+
+    Args:
+      name: Name of the repo rule.
+      dicts: The execution property set constants.
```

Comment:
explain in more detail the expected data structure

Reply:
done